lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,822 @@
+	}
+
+	/**
+	 * Checks if a given request has access to delete a specific plugin.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to delete the item, WP_Error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'activate_plugins' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_plugins',
+				__( 'Sorry, you are not allowed to manage plugins for this site.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( ! current_user_can( 'delete_plugins' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_plugins',
+				__( 'Sorry, you are not allowed to delete plugins for this site.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$can_read = $this->check_read_permission( $request['plugin'] );
+
+		if ( is_wp_error( $can_read ) ) {
+			return $can_read;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deletes one plugin from the site.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$data = $this->get_plugin_data( $request['plugin'] );
+
+		if ( is_wp_error( $data ) ) {
+			return $data;
+		}
+
+		if ( is_plugin_active( $request['plugin'] ) ) {
+			return new WP_Error(
+				'rest_cannot_delete_active_plugin',
+				__( 'Cannot delete an active plugin. Please deactivate it first.', 'gutenberg' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$filesystem_available = $this->is_filesystem_available();
+		if ( is_wp_error( $filesystem_available ) ) {
+			return $filesystem_available;
+		}
+
+		$deleted = delete_plugins( array( $request['plugin'] ) );
+
+		if ( is_wp_error( $deleted ) ) {
+			return $deleted;
+		}
+
+		return new WP_REST_Response( null, 204 );
+	}
+
+	/**
+	 * Prepares the plugin for the REST response.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param mixed           $item    Unmarked up and untranslated plugin data from {@see get_plugin_data()}.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$item   = _get_plugin_data_markup_translate( $item['_file'], $item, false );
+		$marked = _get_plugin_data_markup_translate( $item['_file'], $item, true, false );
+
+		$data = array(
+			'plugin'       => $item['_file'],
+			'status'       => $this->get_plugin_status( $item['_file'] ),
+			'name'         => $item['Name'],
+			'plugin_uri'   => $item['PluginURI'],
+			'author'       => $item['Author'],
```

Comment:
What about 
```
'TextDomain'  => 'Text Domain',
'DomainPath'  => 'Domain Path',
```

Reply:
As per the themes ticket, going to include text domain. Domain path is an internal detail, and I don't think we just want to expose the header for it.

Reply:
Added in a4bb8eb.